hydroflow/src/scheduled/context.rs

```diff
@@ -188,3 +206,40 @@ impl Context {
         futures::future::join_all(self.task_join_handles.drain(..)).await;
     }
 }
+/// Internal APIs.
+impl Context {
+    /// Create a new context for the Hydroflow graph instance, used internally.
+    pub(crate) fn new(event_queue_send: UnboundedSender<(SubgraphId, bool)>) -> Self {
+        Context {
+            states: Vec::new(),
+
+            event_queue_send,
+
+            current_stratum: 0,
+            current_tick: TickInstant::default(),
+
+            current_tick_start: SystemTime::now(),
+            subgraph_last_tick_run_in: None,
+
+            subgraph_id: SubgraphId(0),
+
+            tasks_to_spawn: Vec::new(),
+            task_join_handles: Vec::new(),
+        }
+    }
+
+    pub(crate) fn reset_state_at_end_of_tick(&mut self) {
+        for StateData { state, tick_reset } in self.states.iter_mut() {
+            if let Some(tick_reset) = tick_reset {
+                (tick_reset)(Box::deref_mut(state));
+            }
+        }
+    }
+}
+
+/// Internal struct containing a pointer to [`Hydroflow`]-owned state.
+struct StateData {
+    state: Box<dyn Any>,
+    tick_reset: Option<TickResetFn>,
+}
+type TickResetFn = Box<dyn FnMut(&mut dyn Any)>;
```

Comment:
This could use some documentation.
1. What is the use of a TickResetFn?
2. Who provides it? 
3. Who calls it? When?